events/tests/test_api.py

```diff
@@ -212,3 +272,43 @@ def test_occurrence_query_normal_user(snapshot, user_api_client):
     executed = user_api_client.execute(OCCURRENCE_QUERY, variables=variables)
 
     snapshot.assert_match(executed)
+
+
+def test_add_event_permission_denied(api_client, user_api_client):
+    executed = api_client.execute(ADD_EVENT_MUTATION, variables=ADD_EVENT_VARIABLES)
+    assert_permission_denied(executed)
+
+    executed = user_api_client.execute(
+        ADD_EVENT_MUTATION, variables=ADD_EVENT_VARIABLES
+    )
+    assert_permission_denied(executed)
+
+
+def test_add_event_staff_user(snapshot, staff_api_client):
+    executed = staff_api_client.execute(
+        ADD_EVENT_MUTATION, variables=ADD_EVENT_VARIABLES
+    )
+    snapshot.assert_match(executed)
+
+
+def test_add_occurrence_permission_denied(api_client, user_api_client):
+    executed = api_client.execute(
+        ADD_OCCURRENCE_MUTATION, variables=ADD_OCCURRENCE_VARIABLES
+    )
+    assert_permission_denied(executed)
+
+    executed = user_api_client.execute(
+        ADD_OCCURRENCE_MUTATION, variables=ADD_OCCURRENCE_VARIABLES
+    )
+    assert_permission_denied(executed)
+
+
+def test_add_occurrence_staff_user(snapshot, staff_api_client):
+    event = EventFactory()
+    venue = VenueFactory()
+    ADD_OCCURRENCE_VARIABLES["input"]["eventId"] = to_global_id("EventNode", event.id)
+    ADD_OCCURRENCE_VARIABLES["input"]["venueId"] = to_global_id("VenueNode", venue.id)
```

Comment:
This is a bit bad, because it modifies `ADD_OCCURRENCE_VARIABLES` dict, so running this test will affect other tests as well. So you should (deep)copy the dict first. 

It could probably make sense to use pytest fixtures for variables instead of those dicts, but we can figure that out later :)

Reply:
Thanks.  Didn't think of it because that variable has been used only in this test. I'll update it